Distiller-server/src/main/java/com/mapr/distiller/server/metricactions/DiskstatMetricAction.java

```diff
@@ -0,0 +1,62 @@
+package com.mapr.distiller.server.metricactions;
+
+import java.util.Map;
+
+import com.mapr.distiller.server.processors.DiskstatRecordProcessor;
+import com.mapr.distiller.server.processors.RecordProcessor;
+import com.mapr.distiller.server.queues.RecordQueue;
+import com.mapr.distiller.server.queues.RecordQueueManager;
+import com.mapr.distiller.server.recordtypes.Record;
+import com.mapr.distiller.server.utils.MetricConfig;
+
+public class DiskstatMetricAction extends MetricAction {
+
+	private DiskstatMetricAction(String id, String recordType,
+			String aggregationType, Map<String, String> aggregationMap,
+			RecordQueue inputQueue, RecordQueue outputQueue,
+			RecordProcessor<Record> recordProcessor) {
+		super(id, recordType, aggregationType, aggregationMap, inputQueue,
+				outputQueue, recordProcessor);
+	}
+
+	public static DiskstatMetricAction getInstance(MetricConfig metricConfig,
+			RecordQueueManager queueManager) {
+		RecordProcessor<Record> recordProcessor = new DiskstatRecordProcessor();
+		return new DiskstatMetricAction(metricConfig.getId(),
+				metricConfig.getRecordType(),
+				metricConfig.getAggregationType(),
+				metricConfig.getAggregationMap(),
+				queueManager.getQueue(metricConfig.getInputQueue()),
+				queueManager.getQueue(metricConfig.getOutputQueue()),
+				recordProcessor);
+	}
+
+	@Override
+	public void run() {
```

Comment:
run() looks the same between the various MetricAction extending classes, is there a specific reason it isn't just sourced from MetricAction.java instead of being in each sub class?